src/Stan/Analysis.hs

```diff
@@ -0,0 +1,51 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Static analysis of all HIE files.
+-}
+
+module Stan.Analysis
+    ( Analysis (..)
+    , runAnalysis
+    ) where
+
+import HieTypes (HieFile)
+
+
+{- | This data type stores all information collected during static analysis.
+-}
+data Analysis = Analysis
+    { analysisModuleCount    :: !Int
+    , analysisLinesOfCode    :: !Int
+    , analysisUsedExtensions :: !Int
+    , analysisObservations   :: !()  -- TODO: use Observation type later
+    }
+
+initialAnalysis :: Analysis
+initialAnalysis = Analysis
+    { analysisModuleCount    = 0
+    , analysisLinesOfCode    = 0
+    , analysisUsedExtensions = 0
+    , analysisObservations   = ()
+    }
+
+-- QUESTION: is it better to create a newtype? Also, help with the name...
+type AnalyseM a = State Analysis a
```

Comment:
For the name, I would prefer a more specific one, like `AnalysisState`. But this then raises the question, do we need the type alias/newtype then?

Reply:
@vrom911 Good point :+1: I don't mind expanding type alias explicitly